packages/jupyter-ai/jupyter_ai/extension.py

```diff
@@ -263,3 +263,29 @@ def initialize_settings(self):
 
     async def _get_dask_client(self):
         return DaskClient(processes=False, asynchronous=True)
+
+    async def stop_extension(self):
+        """
+        Public method called by Jupyter Server when the server is stopping.
+        This calls the cleanup code defined in `self._stop_exception()` inside
+        an exception handler, as the server halts if this method raises an
+        exception.
+        """
+        try:
+            await self._stop_extension()
+        except Exception as e:
+            self.log.error(
+                "Jupyter AI raised an exception while stopping, printed below. Please report this to the maintainers on GitHub."
```

Comment:
```suggestion
                "Jupyter AI raised an exception while stopping:"
```

Let's keep the errors minimal and not encourage all exceptions to be reported as issues.